src/request_collector/server.py

```diff
@@ -121,5 +136,4 @@ def on_monitor_request(self, request_monitoring: RequestMonitoring) -> None:
             reward_amount=monitor_request.reward_amount,
         )
 
-        with self.state_db.conn:
-            self.state_db.upsert_monitor_request(monitor_request)
+        self.state_db.upsert_monitor_request(monitor_request)
```

Comment:
Why can we remove the transaction here?

Reply:
It is only a single operation. Those are atomic anyway.